sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -106,51 +111,69 @@ internal static IEnumerable<BatchedLogs<T>> Batch<T>(IEnumerable<T> logEntries,
-                    WriteMemory(tempStream, BinaryData.FromString("]").ToMemory());
-                    tempStream.Position = 0;
-                    yield return new BatchedLogs<T>(new List<T>{log}, BinaryData.FromStream(tempStream));
+                    // Create tempArrayBufferWriter (unsized to store log) and tempWriter for individual log
+                    Azure.Core.ArrayBufferWriter<byte> tempArrayBuffer = new Azure.Core.ArrayBufferWriter<byte>();
+                    Utf8JsonWriter tempWriter = new Utf8JsonWriter(tempArrayBuffer);
+                    tempWriter.WriteStartArray();
+                    WriteMemory(tempWriter, memory);
+                    tempWriter.WriteEndArray();
+                    tempWriter.Flush();
+                    yield return new BatchedLogs<T>(new List<T>{log}, BinaryData.FromBytes(tempArrayBuffer.WrittenMemory));
                 }
-                else if ((stream.Length + memory.Length + 1) >= SingleUploadThreshold) // if adding this entry makes stream > 1 Mb send current stream now
+                // if adding this entry makes stream > 1 Mb send current stream now
+                else if ((writer.BytesPending + memory.Length + 1) >= SingleUploadThreshold)
                 {
-                    WriteMemory(stream, BinaryData.FromString("]").ToMemory());
-                    stream.Position = 0; // set Position to 0 to return everything from beginning of stream
-                    yield return new BatchedLogs<T>(currentLogList, BinaryData.FromStream(stream));
-
-                    // reset stream and currentLogList
-                    stream = new MemoryStream(SingleUploadThreshold); // reset stream
-                    currentLogList = new List<T>(); // reset log list
-                    WriteMemory(stream, memory); // add log to memory and currentLogList
+                    writer.WriteEndArray();
+                    writer.Flush();
+                    // This batch is full so send it now
+                    yield return new BatchedLogs<T>(currentLogList, BinaryData.FromBytes(arrayBuffer.WrittenMemory));
+
+                    // Reset arrayBuffer and writer for next batch
+                    arrayBuffer = new Azure.Core.ArrayBufferWriter<byte>(SingleUploadThreshold);
+                    writer.Reset(arrayBuffer);
+                    writer.WriteStartArray();
+                    // reset log list
+                    currentLogList = new List<T>();
+                    // add current log to memory and currentLogList
+                    WriteMemory(writer, memory);
                     currentLogList.Add(log);
+
+                    // if this is the last log, send batch now
+                    if (isLastEntry)
+                    {
+                        writer.WriteEndArray();
+                        writer.Flush();
+                        yield return new BatchedLogs<T>(currentLogList, BinaryData.FromBytes(arrayBuffer.WrittenMemory));
```

Comment:
Is the logList used anywhere?

Reply:
no - we are only using it for the count so I'll replace the saved logs with the count